packages/nuxt/src/runtime/composables.ts

```diff
@@ -1,5 +1,33 @@
 import { useNuxtApp } from '#app'
-import type { Pinia } from 'pinia'
+import {
+  defineStore as _defineStore,
+  type Pinia,
+  type StoreGeneric,
+} from 'pinia'
 export * from 'pinia'
 
-export const usePinia = () => useNuxtApp().$pinia as Pinia
+export const usePinia = () => useNuxtApp().$pinia as Pinia | undefined
+
+export const defineStore: typeof _defineStore =
+  process.env.NODE_ENV === 'production' && !__TEST__
+    ? _defineStore
+    : (((
+        ...args: Parameters<typeof _defineStore>
+      ): ReturnType<typeof _defineStore> => {
+        if (!import.meta.server) {
```

Comment:
This condition can be probably be moved up so we simply return `_defineStore` too